lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -2521,12 +2521,14 @@ getNetworkInformation st nl = liftIO $ do
 getNetworkParameters
     :: (Block, NetworkParameters, SyncTolerance)
     -> NetworkLayer IO Block
+    -> TransactionLayer k
     -> Handler ApiNetworkParameters
-getNetworkParameters (_block0, genesisNp, _st) nl = do
+getNetworkParameters (_block0, genesisNp, _st) nl tl = do
     pp <- liftIO $ NW.currentProtocolParameters nl
     sp <- liftIO $ NW.currentSlottingParameters nl
     let np = genesisNp { protocolParameters = pp, slottingParameters = sp }
-    liftIO $ toApiNetworkParameters np (interpretQuery ti . toApiEpochInfo)
+    let txConstraints = (view #constraints tl) pp
+    liftIO $ toApiNetworkParameters np txConstraints (interpretQuery ti . toApiEpochInfo)
```

Comment:
Can we not involve TransactionLayer here?
It seems a bit roundabout to call a transaction layer thing that just references the protocol parameters.

Reply:
> It seems a bit roundabout to call a transaction layer thing that just references the protocol parameters.

I fully agree. But to convert the Alonzo `adaPerUTxOWord` to our generic API `minUTxOValue`, we do need to call the `cardano-wallet`-specific (i.e not in `-core`) function `computeMinimumAdaQuantityInternal`…

Alternatives could be:
1. Pass in using some different record/function than the TransactionLayer — but we already have this one
2. Convert `MinUTxOValue` to `Coin` already before storing it in the`ProtocolParameters` record — sounds like a horrible idea
3. Merge `cardano-wallet` and `cardano-wallet-core`

Aside from the merging the libraries, I don't see anything obvious.

Reply:
Oh, that's ... unfortunate. I have opened PR #2787 to fix this.